### ISSUE TITLE
/hideroom prevents the bot from acting

### ISSUE THREAD

**Issue by DiscordOdysseyDev** [pmariglia/showdown#68]
What the title says, if you use the command /hideroom the bot stops acting for the rest of the game (which means after the bot loses because of timeout), i have been trying to do something about this but my knowledge isn't enough to understand how this bot works. I'm still trying to figure how and why and any help will be much appreciated. I'm posting this here because it seems relevant enough.

Steps to reproduce: Just use /hideroom when facing a bot.

**Comment by pmariglia** [comment]
Nice catch, looks like the room gets renamed when someone uses `/hideroom`. The bot needs the current name of the room in order to send messages, and it won't parse the following battle log in order to add the random characters to the end of the room:
```
>battle-gen8randombattle-1372565845
|noinit|rename|battle-gen8randombattle-1372565845-qddmfz5d31w306mbl6odghadmmvzpyjpw
```

Should be a simple fix - just need to write some battle-log parsing to update the name of the room when it sees a message like this. Ill push a fix when I get a chance.

**Comment by DiscordOdysseyDev** [comment]
Thanks for the fast response, I'd be glad to help in anything I can but my python knowledge is very basic. I'll stay tuned for updates

**Comment by pmariglia** [comment]
Should be (mostly) fixed in 5c4add916b639f019c9643efd6a40a1ad119bdcc.

There's probably still a weird edge-case where if `/hideroom` is used while the bot is still thinking. I'll get back to this when I have a bit more time.
